modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/MultiSearchTemplateResponseTests.java

```diff
@@ -0,0 +1,95 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.script.mustache;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.search.ShardSearchFailure;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.internal.InternalSearchResponse;
+import org.elasticsearch.test.AbstractXContentTestCase;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+public class MultiSearchTemplateResponseTests extends AbstractXContentTestCase<MultiSearchTemplateResponse> {
+
+    @Override
+    protected MultiSearchTemplateResponse createTestInstance() {
+        int numItems = randomIntBetween(0, 128);
+        long overallTookInMillis = randomNonNegativeLong();
+        MultiSearchTemplateResponse.Item[] items = new MultiSearchTemplateResponse.Item[numItems];
+        for (int i = 0; i < numItems; i++) {
+            if (randomBoolean()) {
+                // Creating a minimal response is OK, because SearchResponse self
+                // is tested elsewhere.
+                long tookInMillis = randomNonNegativeLong();
+                int totalShards = randomIntBetween(1, Integer.MAX_VALUE);
+                int successfulShards = randomIntBetween(0, totalShards);
+                int skippedShards = totalShards - successfulShards;
+                InternalSearchResponse internalSearchResponse = InternalSearchResponse.empty();
+                SearchResponse.Clusters clusters = new SearchResponse.Clusters(totalShards, successfulShards, skippedShards);
+                SearchTemplateResponse searchTemplateResponse = new SearchTemplateResponse();
+                SearchResponse searchResponse = new SearchResponse(internalSearchResponse, null, totalShards,
+                        successfulShards, skippedShards, tookInMillis, ShardSearchFailure.EMPTY_ARRAY, clusters);
+                searchTemplateResponse.setResponse(searchResponse);
+                items[i] = new MultiSearchTemplateResponse.Item(searchTemplateResponse, null);
+            } else {
+                items[i] = new MultiSearchTemplateResponse.Item(null, new ElasticsearchException("an error"));
+            }
+        }
+        return new MultiSearchTemplateResponse(items, overallTookInMillis);
+    }
+
+    @Override
+    protected MultiSearchTemplateResponse doParseInstance(XContentParser parser) throws IOException {
+        return MultiSearchTemplateResponse.fromXContext(parser);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
+    }
+
+    @Override
+    protected boolean assertToXContentEquivalence() {
+        return false;
```

Comment:
what made you return false here? checking whether this is expected or not. may have to do with exceptions which are parsed differently compared to their original representation. In that case, we could have an ordinary test without exceptions and a variation of it with exceptions where we disabled this check. We do this already in `ListTasksResponseTests`